torchao/float8/fsdp_utils.py

```diff
@@ -59,7 +59,7 @@ def precompute_float8_dynamic_scale_for_fsdp(module: nn.Module) -> None:
         return
 
     # inf-norm is equivalent to max(abs(w))
-    max_weights = torch._foreach_norm(weights, ord=math.inf, dtype=torch.float32)  # Partial
+    max_weights = torch._foreach_norm(weights, ord=math.inf)  # Partial
```

Comment:
wondering why remove `dtype=torch.float32` here / why does it matter?

Reply:
this line is mimicing float8 compute when we do abs(max) without upscaling
https://github.com/pytorch/ao/blob/144445a7a8e988059555421caafa17e0c1678053/torchao/float8/float8_utils.py#L101-L102C28

not sure if it improves groudtruth numerics, but at least this brings numeric on par with float8 compute https://github.com/pytorch/ao/blob/144445a7a8e988059555421caafa17e0c1678053/test/float8/test_fsdp2/test_fsdp2.py#L124-L132